elixir/apps/domain/test/support/new_fixtures/token_fixtures.ex

```diff
@@ -7,56 +7,65 @@ defmodule Domain.TokenFixtures do
   import Domain.ActorFixtures
   import Domain.SiteFixtures
 
-  @doc """
-  Generate valid token attributes with sensible defaults.
-  """
-  def valid_token_attrs(attrs \\ %{}) do
-    unique_num = System.unique_integer([:positive, :monotonic])
-
-    Enum.into(attrs, %{
-      type: :browser,
-      name: "Token #{unique_num}",
-      secret_salt: generate_salt(),
-      secret_hash: generate_hash(),
-      remaining_attempts: 3,
-      # Default expiration 30 days from now
-      expires_at: DateTime.add(DateTime.utc_now(), 30, :day)
-    })
-  end
-
   @doc """
   Generate a token with valid default attributes.
 
-  The token will be created with an associated account and actor/site
-  depending on the token type.
+  Returns the token with `secret_fragment` populated as a virtual field.
+  Use `encode_token/1` or `encode_token/2` to get the encoded string.
+
+  ## Options
+
+    * `:type` - Token type (default: `:browser`)
+    * `:account` - Account to associate with the token
+    * `:actor` - Actor for browser/client/api_client tokens
+    * `:site` - Site for site tokens
+    * `:secret_fragment` - Custom secret fragment (generated if not provided)
+    * `:nonce` - Nonce for hash computation (default: "")
+    * `:expires_at` - Token expiration time
 
   ## Examples
 
       token = token_fixture()
-      token = token_fixture(type: :client)
-      token = token_fixture(actor: actor, type: :browser)
+      token = client_token_fixture()
+      encoded = encode_token(token)
 
   """
   def token_fixture(attrs \\ %{}) do
     attrs = Enum.into(attrs, %{})
 
-    # Get or create account
-    account = Map.get(attrs, :account) || account_fixture()
-
-    # Build token attrs
-    token_attrs =
-      attrs
-      |> Map.delete(:account)
-      |> Map.delete(:actor)
-      |> Map.delete(:site)
-      |> Map.delete(:auth_provider)
-      |> valid_token_attrs()
+    type = Map.get(attrs, :type, :browser)
+    name = Map.get(attrs, :name, "Token #{System.unique_integer([:positive])}")
+    remaining_attempts = Map.get(attrs, :remaining_attempts, 0)
+    nonce = Map.get(attrs, :nonce, "")
+    expires_at = Map.get(attrs, :expires_at, DateTime.add(DateTime.utc_now(), 30, :day))
+    secret_fragment = Map.get(attrs, :secret_fragment, generate_secret_fragment())
+    secret_salt = generate_salt()
+    secret_hash = compute_secret_hash(nonce, secret_fragment, secret_salt)
+
+    token_attrs = %{
+      type: type,
+      name: name,
+      secret_nonce: nonce,
+      secret_salt: secret_salt,
+      secret_hash: secret_hash,
+      remaining_attempts: remaining_attempts,
+      expires_at: expires_at
+    }
```

Comment:
Any reason not to use the `valid_token_attrs()` function like the other fixtures?